### ISSUE TITLE
Porting SqueezeNet to Facenet training accuracy stuck

### ISSUE THREAD

**Issue by llj098** [davidsandberg/facenet#92]
Hello @davidsandberg, 

I am trying to add [SqueezeNet](https://github.com/DeepScale/SqueezeNet) to your FaceNet implementation. My code is (models/squeezenet.py):
```python
from __future__ import absolute_import
from __future__ import division
from __future__ import print_function

import tensorflow as tf
from tensorflow.contrib.framework import add_arg_scope
from tensorflow.contrib.layers.python.layers import utils
slim = tf.contrib.slim

def squeeze(inputs, num_outputs, fire_id):
    return slim.conv2d(inputs, num_outputs, 1, stride=1, scope="fire/squeeze/"+str(fire_id))

def expand(inputs, num_outputs, fire_id):
    with tf.variable_scope('expand'):
        e1x1 = slim.conv2d(inputs, num_outputs, 1, stride=1, scope='fire/ex1x1/'+str(fire_id))
        e3x3 = slim.conv2d(inputs, num_outputs, 3, scope='fire/ex3x3/'+str(fire_id))
    return tf.concat(3, [e1x1, e3x3])

def fire_module(x, s=16, e=64, fire_id=0):
    return expand(squeeze(x, s, fire_id), e, fire_id)

def inference(images, _a, phase_train=True, weight_decay=0.8, reuse=None):
    with tf.variable_scope('SqueezeNet', 'SqueezeNet', [images], reuse=reuse):
            x_image = tf.reshape(images, [-1, 160, 160, 3])
            net = slim.conv2d(x_image, 96, 7, scope="conv_1", stride=2) 
            net = slim.max_pool2d(net, 3, stride=2, scope="maxpool1") 
            net = fire_module(net, fire_id=2)
            net = fire_module(net, fire_id=3)
            net = fire_module(net, 32, 128, fire_id=4)
            net = slim.max_pool2d(net, 3, scope="maxpool3")
            net = fire_module(net, 32, 128, fire_id=5)
            net = fire_module(net, 48, 192, fire_id=6)
            net = fire_module(net, 48, 192, fire_id=7)
            net = fire_module(net, 64, 256, fire_id=8)
            net = slim.max_pool2d(net, 2, stride=2, scope="maxpool8")
            net = fire_module(net, 64, 256, fire_id=9)
            net = slim.dropout(net, 0.5)
            net = slim.conv2d(net, 1000, 1, scope="conv10", padding="VALID")
            net = slim.avg_pool2d(net, 9)
            net = slim.flatten(net)
    return net,{}
```

When I run `facenet_train_classifier.py` with  `--model_def models.squeezenet`, everything works, but, after about 50 epochs, I got `accuracy: 0.485` on validation in every epoch.

I am new to machine learning(also tensorflow), could you please help me to find out why the accuracy does not increase ? (After this, I will send you a pull request  😄  )

Thanks in advance! 

---

PS: 

1. I have already tried this implementation with mnist data set, the accuracy result on test set is : 98.73%
2. git head is e5d852a3eb7a080f0a8006d281d6d9175200a614

**Comment by llj098** [comment]
The full command is :

```
 python facenet_train_classifier.py --logs_base_dir ~/logs/facenet/ --models_base_dir ~/models/facenet/ --data_dir ~/facenet/data/train_test/facescrub_mtcnnalign_182_160:~/facenet/data/casia_maxpy_mtcnnpy_182/  --image_size 160 --model_def models.squeezenet  --lfw_dir ~/facenet/data/lfw_mtcnnpy_160 --weight_decay 2e-4 --optimizer RMSPROP --learning_rate -1 --max_nrof_epochs 80 --keep_probability 0.8 --random_crop --random_flip --learning_rate_schedule_file ../data/learning_rate_schedule_classifier_long.txt --center_loss_factor 2e-4
```


The log is :

```
...

Batch 117 in 0.409 seconds
Batch 118 in 0.428 seconds
Batch 119 in 0.411 seconds
Accuracy: 0.485+-0.008
Validation rate: 1.00000+-0.00000 @ FAR=1.00000
Epoch: [56][1/1000]     Time 0.276      Loss 8.857      RegLoss 0.008
Epoch: [56][2/1000]     Time 0.202      Loss 9.090      RegLoss 0.008
Epoch: [56][3/1000]     Time 0.199      Loss 8.798      RegLoss 0.008
Epoch: [56][4/1000]     Time 0.200      Loss 9.007      RegLoss 0.008
Epoch: [56][5/1000]     Time 0.202      Loss 8.869      RegLoss 0.008
Epoch: [56][6/1000]     Time 0.201      Loss 8.812      RegLoss 0.008
Epoch: [56][7/1000]     Time 0.203      Loss 8.794      RegLoss 0.008
...
```

**Comment by davidsandberg** [comment]
Closing this since I will not have time to look at it.
